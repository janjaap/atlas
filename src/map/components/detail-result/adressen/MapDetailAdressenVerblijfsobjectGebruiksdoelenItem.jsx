import React from 'react';
import PropTypes from 'prop-types';

import { maxDisplayValuesPerProperty } from '../../../services/map-detail';

const MapDetailAdressenVerblijfsobjectGebruiksdoelenItem = ({ gebruiksdoelen }) =>
  gebruiksdoelen.length > 0 && (
    <div className="map-detail-result__item">
      <section className="map-detail-result__item-content">
        <div className="map-detail-result__item-label">Gebruiksdoel</div>
        <ul className="map-detail-result__item-value map-detail-result__item-list">
          {gebruiksdoelen.slice(0, maxDisplayValuesPerProperty).map((item) => (
            <li
              key={item.code}
              className="map-detail-result__item-list-item"
            >
              {item.description}
              {item.descriptionPlus && (
                <span>: {item.descriptionPlus}</span>
              )}
            </li>
          ))}
        </ul>
      </section>
    </div>
  );

MapDetailAdressenVerblijfsobjectGebruiksdoelenItem.propTypes = {
  gebruiksdoelen: PropTypes.arrayOf(PropTypes.shape({
    code: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    descriptionPlus: PropTypes.string
  })).isRequired
};

export default MapDetailAdressenVerblijfsobjectGebruiksdoelenItem;
