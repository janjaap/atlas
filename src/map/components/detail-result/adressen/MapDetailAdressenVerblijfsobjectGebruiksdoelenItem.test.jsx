import React from 'react';
import { shallow } from 'enzyme';

import MapDetailAdressenVerblijfsobjectGebruiksdoelenItem from './MapDetailAdressenVerblijfsobjectGebruiksdoelenItem';

describe('MapDetailAdressenVerblijfsobjectGebruiksdoelenItem', () => {
  it('should render no more than five gebruiksdoelen', () => {
    const gebruiksdoelen = [{
      code: '01',
      description: 'Description 1'
    }, {
      code: '02',
      description: 'Description 2',
      descriptionPlus: 'Description 2 Plus'
    }, {
      code: '03',
      description: 'Description 3'
    }, {
      code: '04',
      description: 'Description 4'
    }, {
      code: '05',
      description: 'Description 5',
      descriptionPlus: 'Description 5 Plus'
    }, {
      code: '06',
      description: 'Description 6'
    }];
    const wrapper = shallow(
      <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
        gebruiksdoelen={gebruiksdoelen}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render exactly five gebruiksdoelen', () => {
    const gebruiksdoelen = [{
      code: '01',
      description: 'Description 1'
    }, {
      code: '02',
      description: 'Description 2',
      descriptionPlus: 'Description 2 Plus'
    }, {
      code: '03',
      description: 'Description 3'
    }, {
      code: '04',
      description: 'Description 4'
    }, {
      code: '05',
      description: 'Description 5',
      descriptionPlus: 'Description 5 Plus'
    }];
    const wrapper = shallow(
      <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
        gebruiksdoelen={gebruiksdoelen}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render two gebruiksdoelen', () => {
    const gebruiksdoelen = [{
      code: '01',
      description: 'Description 1',
      descriptionPlus: 'Description 1 Plus'
    }, {
      code: '02',
      description: 'Description 2'
    }];
    const wrapper = shallow(
      <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
        gebruiksdoelen={gebruiksdoelen}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one gebruiksdoel', () => {
    const gebruiksdoelen = [{
      code: '01',
      description: 'Description 1'
    }];
    const wrapper = shallow(
      <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
        gebruiksdoelen={gebruiksdoelen}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render no gebruiksdoelen', () => {
    const gebruiksdoelen = [];
    const wrapper = shallow(
      <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
        gebruiksdoelen={gebruiksdoelen}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
