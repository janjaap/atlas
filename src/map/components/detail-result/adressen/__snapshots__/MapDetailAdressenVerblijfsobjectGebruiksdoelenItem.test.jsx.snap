// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailAdressenVerblijfsobjectGebruiksdoelenItem should render exactly five gebruiksdoelen 1`] = `
<div
  className="map-detail-result__item"
>
  <section
    className="map-detail-result__item-content"
  >
    <div
      className="map-detail-result__item-label"
    >
      Gebruiksdoel
    </div>
    <ul
      className="map-detail-result__item-value map-detail-result__item-list"
    >
      <li
        className="map-detail-result__item-list-item"
        key="01"
      >
        Description 1
      </li>
      <li
        className="map-detail-result__item-list-item"
        key="02"
      >
        Description 2
        <span>
          : 
          Description 2 Plus
        </span>
      </li>
      <li
        className="map-detail-result__item-list-item"
        key="03"
      >
        Description 3
      </li>
      <li
        className="map-detail-result__item-list-item"
        key="04"
      >
        Description 4
      </li>
      <li
        className="map-detail-result__item-list-item"
        key="05"
      >
        Description 5
        <span>
          : 
          Description 5 Plus
        </span>
      </li>
    </ul>
  </section>
</div>
`;

exports[`MapDetailAdressenVerblijfsobjectGebruiksdoelenItem should render no gebruiksdoelen 1`] = `""`;

exports[`MapDetailAdressenVerblijfsobjectGebruiksdoelenItem should render no more than five gebruiksdoelen 1`] = `
<div
  className="map-detail-result__item"
>
  <section
    className="map-detail-result__item-content"
  >
    <div
      className="map-detail-result__item-label"
    >
      Gebruiksdoel
    </div>
    <ul
      className="map-detail-result__item-value map-detail-result__item-list"
    >
      <li
        className="map-detail-result__item-list-item"
        key="01"
      >
        Description 1
      </li>
      <li
        className="map-detail-result__item-list-item"
        key="02"
      >
        Description 2
        <span>
          : 
          Description 2 Plus
        </span>
      </li>
      <li
        className="map-detail-result__item-list-item"
        key="03"
      >
        Description 3
      </li>
      <li
        className="map-detail-result__item-list-item"
        key="04"
      >
        Description 4
      </li>
      <li
        className="map-detail-result__item-list-item"
        key="05"
      >
        Description 5
        <span>
          : 
          Description 5 Plus
        </span>
      </li>
    </ul>
  </section>
</div>
`;

exports[`MapDetailAdressenVerblijfsobjectGebruiksdoelenItem should render one gebruiksdoel 1`] = `
<div
  className="map-detail-result__item"
>
  <section
    className="map-detail-result__item-content"
  >
    <div
      className="map-detail-result__item-label"
    >
      Gebruiksdoel
    </div>
    <ul
      className="map-detail-result__item-value map-detail-result__item-list"
    >
      <li
        className="map-detail-result__item-list-item"
        key="01"
      >
        Description 1
      </li>
    </ul>
  </section>
</div>
`;

exports[`MapDetailAdressenVerblijfsobjectGebruiksdoelenItem should render two gebruiksdoelen 1`] = `
<div
  className="map-detail-result__item"
>
  <section
    className="map-detail-result__item-content"
  >
    <div
      className="map-detail-result__item-label"
    >
      Gebruiksdoel
    </div>
    <ul
      className="map-detail-result__item-value map-detail-result__item-list"
    >
      <li
        className="map-detail-result__item-list-item"
        key="01"
      >
        Description 1
        <span>
          : 
          Description 1 Plus
        </span>
      </li>
      <li
        className="map-detail-result__item-list-item"
        key="02"
      >
        Description 2
      </li>
    </ul>
  </section>
</div>
`;
